internal/broker/broker.go

```diff
@@ -366,24 +364,27 @@ func (b *Broker) IsAuthenticated(sessionID, authenticationData string) (string,
 	defer b.CancelIsAuthenticated(sessionID)
 
 	authDone := make(chan struct{})
-	var access, data string
+	var result isAuthenticatedResult
 	go func() {
-		access, data = b.handleIsAuthenticated(ctx, &session, authData)
+		result = b.handleIsAuthenticated(ctx, &session, authData)
 		close(authDone)
 	}()
 
 	select {
 	case <-authDone:
 	case <-ctx.Done():
-		return AuthCancelled, `{"message": "authentication request cancelled"}`, ctx.Err()
+		result := isAuthenticatedResult{Message: "authentication request cancelled"}
+		data, _ := result.toJSON()
+		return data, AuthCancelled, ctx.Err()
```

Comment:
Returned values are out of order...

Reply:
Uuups, yeah... Reason why I was using a single function, to avoid this kind of errors that are quite easy to get since the current code, that is not strictly typed using strings, is fragile to this kind of issues